test/NotifierTestHelper.h

```diff
@@ -109,42 +109,12 @@ bool SleepUntilCondition(std::vector<std::shared_ptr<std::atomic<bool>>> conditi
    return true;
 }
 
-bool SleepUntilConditionIsFalse(std::vector<std::shared_ptr<std::atomic<bool>>> conditions) {
-   StopWatch timer;
```

Comment:
This was dead code. 

With gcc5x dead code can be detected much easier with the compiler. (that was broken in gcc4). 
With -Werror all warnings are treated as errors so that's why the code was removed (also dead code should be removed)